cmd/internal/ecs/add.go

```diff
@@ -346,14 +386,227 @@ func getClusterState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowS
+		return awf_error(errors.New("Error while loading ECS task definition; please contact support@akitasoftware.com for assistance."))
+	}
+	wf.ecsTaskDefinition = output
+	wf.ecsTaskDefinitionARN = arn(aws.ToString(output.TaskDefinitionArn))
+
+	return awf_next(getServiceState)
+}
+
+func (wf *AddWorkflow) loadServiceFromFlag() (nextState optionals.Optional[AddWorkflowState], err error) {
+	if strings.HasPrefix(ecsServiceFlag, "arn:") {
+		service, err := wf.getService(arn(ecsServiceFlag))
+		if err != nil {
+			return awf_error(errors.Wrap(err, "Error accessing service"))
+		}
+		wf.ecsService = aws.ToString(service.ServiceName)
+		wf.ecsServiceARN = arn(ecsServiceFlag)
+		return awf_next(confirmState)
+	}
+
+	services, listErr := wf.listECSServices()
+	if listErr != nil {
+		return awf_error(errors.Wrap(err, "Error listing services"))
+	}
+	for a, name := range services {
+		if name == ecsServiceFlag {
+			printer.Infof("Found service %q matching name %q.\n", a, name)
+			wf.ecsServiceARN = a
+			wf.ecsService = name
+			return awf_next(confirmState)
+		}
+	}
+	return awf_error(fmt.Errorf("No service found with name %q that uses task definition %q", ecsServiceFlag, wf.ecsTaskDefinitionFamily))
+}
+
+// Find all services in the cluster that match the task definition.
+func getServiceState(wf *AddWorkflow) (nextState optionals.Optional[AddWorkflowState], err error) {
+	reportStep("Get ECS Service")
+
+	if ecsServiceFlag != "" {
+		return wf.loadServiceFromFlag()
+	}
+
+	services, listErr := wf.listECSServices()
+	if listErr != nil {
+		var uoe UnauthorizedOperationError
+		if errors.As(listErr, &uoe) {
+			printer.Errorf("The provided credentials do not have permission to perform %s in the cluster %q.\n",
+				uoe.OperationName, wf.ecsCluster)
+			printer.Infof("Please choose a different cluster, or assign this permission in AWS IAM.\n")
+			return awf_next(getClusterState)
+		}
+		printer.Errorf("Could not list ECS services: %v\n", listErr)
+		return awf_error(errors.New("Error while listing ECS services; try using the --service flag instead."))
+	}
+
+	if len(services) == 0 {
+		printer.Errorf("Could not find any ECS tasks in cluster %q that use task %q. Please select a different task or hit Ctrl+C to exit.\n",
+			wf.ecsCluster, wf.ecsTaskDefinitionFamily)
+		return awf_next(getTaskState)
+	}
+
+	printer.Infof("Found %d services in cluster %q with task definition %q.\n", len(services), wf.ecsCluster, wf.ecsTaskDefinitionFamily)
+
+	choices := make([]string, 0, len(services))
+	for c, _ := range services {
+		choices = append(choices, string(c))
+	}
+	sort.Strings(choices)
+	choices = append(choices, goBackOption)
+	// TODO: allow skipping this step?
+
+	var serviceAnswer string
+	err = survey.AskOne(
+		&survey.Select{
+			Message: "Which service should be restarted to use the modified task?",
+			Help:    "Select ECS service that will be updated with the modified definition, so it can be monitored by Akita.",
```

Comment:
The task definition in the service will switch from "foo:3" to "foo:4", which is a modification. This will cause ECS to spin up a new task with the new definition, and then tear down the old task, which I've glossed as "restart" but I don't know what the idiomatic word is.

https://docs.aws.amazon.com/AmazonECS/latest/userguide/update-service-console-v2.html
> If you want to use an updated container image for your tasks, you can create a new task definition revision with that image and deploy it to your service by using the force new deployment option in the console.

It might be better to use "update" consistently, I'll give that a try.